frb_example/benches/dart/lib/benchmark/strings.dart

```diff
@@ -0,0 +1,52 @@
+// Import BenchmarkBase class.
+import 'package:benchmark_harness/benchmark_harness.dart';
+import 'package:flutter_rust_bridge/flutter_rust_bridge.dart';
+import 'package:flutter_rust_bridge_benchmark/env/stub.dart';
+import 'package:flutter_rust_bridge_benchmark/utils.dart';
+import '../ffi.io.dart' if (dart.library.html) '../ffi.web.dart';
+
+// Create a new benchmark by extending BenchmarkBase
+class TemplateBenchmark extends AsyncBenchmarkBase {
+  TemplateBenchmark() : super('Vec of Strings');
+
+  late FlutterRustBridgeExampleBenchmarkSuiteImpl api;
+  late List<String> strings;
+
+  static void main() {
+    TemplateBenchmark().report();
+  }
+
+  // The benchmark code.
+  @override
+  Future<void> run() async {
+    // ignore: no_leading_underscores_for_local_identifiers
+    final _ = await api.handleStrings(strings: strings);
```

Comment:
Worried whether the output will be optimized away...

Reply:
Feel free to open a PR to improve it in the future ;)

Reply:
>  I would have Rust report me many 0ns long op

https://doc.rust-lang.org/std/hint/fn.black_box.html does this work? (it is nightly, so may need to find some stable equiv)

Reply:
if you check you'll see that mostly I already do something similar with what a black box does: I've setup all my seeds outside of the call site, with random data, and outside of the metrics record span. why do you want to do more complicated than this ?

Reply:
I am talking about "reverse the list" in api.rs, not this optimization, indeed. Because the `list.reverse` is still costly, therefore the time is biased and frb looks very slow even if it is fast.

Reply:
Ah I think I see what you mean now.

You know, as a side note, originally my intent was to get a rough sense of how costly the overhead of sending through FFI is. Especially I was curious to see how the different ways to send over FFI would affect performance, and so I didn't really mind if the metrics are not uber accurate since mostly I was curious to compare the benchmark results between one another. So what I was looking at is not "does this particular **really** takes XXX milliseconds" but more like "how does this particular function which takes a XXX milliseconds compares with this another function which takes YYY milliseconds". When comparing this way, it doesn't really matter if one function takes XXX milliseconds or XXXX milliseconds", since if the metrics are not perfectly accurate, anyway the variance between both is.

Reply:
I totally agree. The problem is, "I was curious to compare the benchmark results between one another" may not be achieved. For example, suppose the real execution for two kinds of things, say your `Vec<Uuid>` optimization, are 20ns+/-2ns vs 10ns+-/1ns (you know, 10% random error), then we know it is twice faster and your optimization is very great. However, the overhead is, for example, 1000ns+/-100ns (again 10% random error). Then the results are almost not distinguishable, and we wrongly think that optimization is useless.

Reply:
Yes do you know that I was actually stroke by the results.

> For example, suppose the real execution for two kinds of things, say your Vec<Uuid> optimization, are 20ns+/-2ns vs 10ns+-/1ns (you know, 10% random error), then we know it is twice faster and your optimization is very great. However, the overhead is, for example, 1000ns+/-100ns (again 10% random error). Then the results are almost not distinguishable, and we wrongly think that optimization is useless.

Yes I had this in mind too. Generally speaking I personally never take any benchmark for more than a simple hint. No matter how well engineered it is, benchmarks themselves depends so much on a tons of factors (which machine, which CPU family, 32 or 64 bits, are there any other processes running simultaneously, so on and so forth) that results can widely vary between executions anyway.

What I was interested in my initial approach was, for example, to find outliers among similar types. Let's say if sending 1,000 `i64`s results in being noticeably faster than, let's say sending 1,000 `u64` then no matter the benchmarks are inaccurate or not, and provided both `i64` and `u64` were benched in a strictly identical way (in terms of call site, initialization, etc) than one can prolly deduce that there's something wrong with e.g. `u64` implementation. Because generally speaking, no matter both were not benchmarked in ideal conditions, the variance of speed between them can somehow be trusted anyway. In other terms, no matter sending 1,000 `i64`s is not e.g. `2ms` but in reality `500us`, and sending 1,000 `u64`s is not `1ms` but in reality `250us`, it doesn't really matter because we can see that sending 1,000 `i64`s is twice as slow as sending 1,000 `u64`s and that gives us a hint that maybe `u64`s internal implementation should be optimized. Something like that.

I was also, as you might guess, curious to see if my optimization for `uuid` was worth it or not. And looking at the results you can see that it's actually slower by a lot than just sending simple `string`. As you rightfully pointed out, I should nevertheless improve my benchmark because the current outcome is far from being trustworthy.

Reply:
> Yes do you know that I was actually stroke by the results.

I also feel weird that your previous optimization did not work indeed.

> Generally speaking I personally never take any benchmark for more than a simple hint. No matter how well engineered it is, benchmarks themselves depends so much on a tons of factors (which machine, which CPU family, 32 or 64 bits, are there any other processes running simultaneously, so on and so forth) that results can widely vary between executions anyway.

Yes benchmarks are very fragile. But anyway they can be quite stable as well - for example, in Flutter repository, they run tons of benchmark for each code commit (!). If performance regresses, they will look at it and see what happens.

> Let's say if sending 1,000 i64s results in being noticeably faster than, let's say sending 1,000 u64 then no matter the benchmarks are inaccurate or not, and provided both i64 and u64 were benched in a strictly identical way (in terms of call site, initialization, etc) than one can prolly deduce that there's something wrong with e.g. u64 implementation.

Agree that is quite helpful case. But problem happens when noise is much larger than the real signal. Say, it takes 10us vs 20us to send u64 vs i64 (so really terrible 200% performance drop and we should fix it), but things like `StopWatch.start()/.stop()` and various things take 1000us. Then you see 1010+-100us vs 1020+-100us, and nobody can realize it is twice slower/faster. In other words, the noise is constant and is large, shadowing the real intersteing thing.

> I was also, as you might guess, curious to see if my optimization for uuid was worth it or not. And looking at the results you can see that it's actually slower by a lot than just sending simple string. As you rightfully pointed out, I should nevertheless improve my benchmark because the current outcome is far from being trustworthy.

No worries, when one day the feature is completed we will all know that :)

Reply:
> I also feel weird that your previous optimization did not work indeed.

I assume it's the overhead of parsing the bytes back into a `UuidValue`.

> (...) In other words, the noise is constant and is large, shadowing the real intersteing thing.

Agreed! 

> No worries, when one day the feature is completed we will all know that :)

Hehe yes

Reply:
> I assume it's the overhead of parsing the bytes back into a UuidValue.

Or, maybe it really indeed worked (i.e. your optimization is great), just shadowed by noise.